tests/misc_env/sosreport.py

```diff
@@ -0,0 +1,67 @@
+""" Collecting logs using sosreport from all nodes in cluster Except Client Node"""
+
+import logging
+import re
+
+from ceph.parallel import parallel
+
+log = logging.getLogger(__name__)
+
+
+def run(**kw):
+    """
+    :param kw:
+       - ceph_nodes: ceph node list representing a cluster
+    :return: 0 on success, 1 for failures
+    """
+    ceph_nodes = kw.get("ceph_nodes")
+    results = dict()
+    log.info("Running sosreport test")
+    with parallel() as p:
+        for cnode in ceph_nodes:
+            if cnode.role != "client":
+                p.spawn(generate_sosreport, cnode, results)
+    log.info(results)
+    if False in results.values():
+        return 1
+    else:
+        return 0
+
+
+def generate_sosreport(cnode, results):
+    """
+    :param:
+       - ceph_nodes: ceph node list representing a cluster
+       - results: Dictionary to store cnode and result of this function as boolean. True for Success False otherwise
+    """
+    node = cnode.hostname
+    results.update({node: True})
+    out, err = cnode.exec_command(cmd="sudo yum -y install sos", long_running=True)
```

Comment:
Use the key `sudo`, i.e. `cnode.exec_command(sudo=True, cmd="yum -y install sos --nogpgcheck")`